src/custom/connection/index.ts

```diff
@@ -47,52 +60,88 @@ export const injectedConnection: Connection = {
   type: ConnectionType.INJECTED,
 }
 
-const [web3GnosisSafe, web3GnosisSafeHooks] = initializeConnector<GnosisSafe>((actions) => new GnosisSafe({ actions }))
+const [web3GnosisSafe, web3GnosisSafeHooks] = initializeConnector<Connector>((actions) => {
+  return new AsyncConnector(
+    () => {
+      return import('@web3-react/gnosis-safe').then((module) => {
+        return new module.GnosisSafe({ actions })
+      })
+    },
+    actions,
+    onError
```

Comment:
Why do you need to pass `action` twice?
I understand the line `70` is to satisfy the original constructor.
As for line `67`, could it be fetched from within the instance?

Not a big deal though